### PR TITLE
Update Hyperopt for TF2 Porting

### PR DESCRIPTION
**Pull request by kaushikb11**
- Remove the Close Session logic for Tf2
- Fix Hyperopt Integration Tests
- Update logic for ParallelExecutor for hyperopt for the Tf2 porting.

### CODE REVIEW

**Comment by w4nderlust** [comment]
This looks good to me so far. Just let's make sure it is doing the right thing with the pid / nvidia-smi.

**Comment by kaushikb11** [comment]
@w4nderlust : This is for three GPUs and 6 workers (ParallelExecutor). Params `gpus` and `gpu_memory_limit` weren't being passed to the Model Class and added support for `gpus` of type `str` in the `initialize_tensorflow` [function](https://github.com/uber/ludwig/blob/tf2_porting/ludwig/utils/tf_utils.py#L48). 
mul_gpus

**Comment by w4nderlust** [comment]
This looks good now, thank you!

**Comment by w4nderlust** [comment]
@tgaddair just to confirm the tensorflow initialization
@calio for confirming this is fine using fiber too

File: ludwig/utils/tf_utils.py [hunk 1]
```diff
@@ -69,7 +69,12 @@ def initialize_tensorflow(gpus=None,
 
     if horovod is not None and gpus is None:
         gpus = [horovod.local_rank()]
-    gpus = [gpus] if isinstance(gpus, int) else gpus
+
+    if isinstance(gpus, int):
+        gpus = [gpus]
+    else:
```
Comment (tgaddair) [inline ludwig/utils/tf_utils.py:75]: This will not work with line 71 above.  I would suggest:

```
elif isintance(gpus, str):
    ...
```

**Comment by w4nderlust** [comment]
> Just one minor issue. Also, does this PR conflict with [3609963](https://github.com/uber/ludwig/commit/36099634bcc84eab4516b1584523aea4d922777d)? Can we do anything to make the merging less painful if so?

I think it shouldn't conflict because I first merged hyperopt in master and then merged master in tf2_porting, and this PR is on tf2_porting.

File: ludwig/utils/tf_utils.py (hunk 1 above)
Comment (w4nderlust) [reply ludwig/utils/tf_utils.py:75]: good catch

**Comment by calio** [comment]
This looks good to me and should not affect FiberExecutor. As long as it works with SerialExecutor, it should work with FiberExecutor.

**Pull request merged**